src/sqlfluff/rules/L040.py

```diff
@@ -15,6 +15,40 @@ class Rule_L040(Rule_L010):
     """Inconsistent capitalisation of boolean/null literal.
 
     The functionality for this rule is inherited from :obj:`Rule_L010`.
+
+    | **Anti-pattern**
+    | In this example, 'null' and 'false' is in lower-case whereas 'TRUE' is in upper-case.
+
+    .. code-block:: sql
+
+        select
+            a,
+            null,
+            TRUE,
+            false
+        from foo
+
+    | **Best practice**
+    | Make all literal null/true/false either in upper-case or in lower-case
```

Comment:
It's not just lower and upper as there is "capitalise" (First letter only capitalised)

```suggestion
    | Ensure all literal null/true/false literals cases are used consistently
```